spec/docker-compose-windows.yml

```diff
@@ -0,0 +1,7 @@
+version: "3"
+services:
+  windows_node:
+    build:
+      context: .
+      dockerfile: Dockerfile.windows_spec
+    container_name: windows_node
```

Comment:
Building a new container is unnecessary here. Instead, just override the `entrypoint` value in compose. Also note, you'll generally want to pick the version based on current "release version" to match the host kernel, which I've made possible by parameterizing with `WIN_RELEASE` (with a reasonable fallback)


```yml
version: "3"
services:
  windows_node:
    image: ${WIN_RELEASE:-mcr.microsoft.com/windows/servercore:1909}
    entrypoint: ['powershell.exe', '-NoLogo', '-NoProfile', '-Command', 'Start-Sleep -Seconds 2147483']
```

If AppVeyor has already cached `microsoft/windowsservercore:latest`, then maybe you just want to set that value in AppVeyor to make things more expedient (note that it doesn't work outside AppVeyor). But if you want to set the version dynamically in AppVeyor, you can use their [Set-AppveyorBuildVariable cmdlet](https://www.appveyor.com/docs/build-worker-api/#set-environment-variable) like:

```powershell
Set-AppveyorBuildVariable -Name 'WIN_RELEASE' -Value ("mcr.microsoft.com/windows/servercore:" + (Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").ReleaseId)
```

FWIW, I did a quick test on a 1909 host hosted in Azure, using a fallback of 1903 and 1909 and it worked fine (but the initial `pull` was sloooooowww which is probably why AppVeyor caches it):

```
PS C:\test> docker-compose up
Pulling windows_node (mcr.microsoft.com/windows/servercore:1903)...
1903: Pulling from windows/servercore
5b663e3b9104: Pull complete                                                                                             c294767ee291: Pull complete                                                                                             Digest: sha256:4fb5944b9ebefeab481893d54a0498a380432f8f80f62f2b9debe81dbd7ff6a7
Status: Downloaded newer image for mcr.microsoft.com/windows/servercore:1903
Creating test_windows_node_1 ... done 
```

Reply:
Updated and added comments